.github/workflows/go-integration.yml

```diff
@@ -0,0 +1,89 @@
+name: Go End-to-End
+
+on:
+  push:
+    branches-ignore:
+      - master
+  pull_request:
+    branches-ignore:
+      - gh-pages
+
+env:
+  GOPATH: /actions-runner/go
+  GOCACHE: /actions-runner/gocache
+  GITHUB_WORKSPACE: /actions-runner/go/src/github.com/G-Research/armada
+  HOME: /actions-runner
+
+jobs:
+  start-runner:
+    name: Start runner in EC2
+    if: github.repository == 'jayofdoom/armada'
+    runs-on: ubuntu-latest
+    outputs:
+      runs-on: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ami-0d3d3ae42ef82e511
```

Comment:
Should these be secrets also?

Reply:
Explicitly not imo, for a handful of reasons:
1) it's not actually secret
2) it's easier to test/change/upgrade images using PRs (e.g. you can PR to test a new image)
3) troubleshooting job failures is painful if you're not certain what base image is being used